pkg/execution/executor/executor.go

```diff
@@ -597,22 +597,28 @@ func (e *executor) Schedule(ctx context.Context, req execution.ScheduleRequest)
 		}
 	}
 
-	err := e.smv2.Create(ctx, newState)
+	st, err := e.smv2.Create(ctx, newState)
 	switch err {
-	case nil:
-		// no-op, continue
-	case state.ErrIdentifierExists:
-		return nil, err
+	case nil, state.ErrIdentifierExists:
+		// no-op: this is safe, continue
 	default:
 		return nil, fmt.Errorf("error creating run state: %w", err)
 	}
+	// override the runID from the state if it doesn't match.
+	// this can happen on scheduling retries due to errors like networking, or abrupt failures
+	//
+	// TODO: update the code from here on to reference the `state` returned instead of using
+	// previous values to avoid weird mismatch issues.
+	// right now, the only mismatch should just be the runID
+	if metadata.ID.RunID != st.RunID() {
```

Comment:
We should be able to replace the metadata wholesale, in every case, because e.smv2.Create either returns `metadata` or the new details, right?

Reply:
we can, it's just the metadata structure is different where the one here that's created and returned is the v2 shape, while the state holds it in the v1 shape.
there are currently no way to convert v1 to v2 shape afaik.